web/packages/teleterm/src/ui/DocumentTerminal/DocumentTerminal.tsx

```diff
@@ -43,10 +44,28 @@ export function DocumentTerminal(props: {
   const { visible, doc } = props;
   const { attempt, initializePtyProcess } = useDocumentTerminal(doc);
   const { upload, download } = useTshFileTransferHandlers();
+  const [showSearch, setShowSearch] = useState(false);
   const unsanitizedTerminalFontFamily = configService.get(
     'terminal.fontFamily'
   ).value;
   const terminalFontSize = configService.get('terminal.fontSize').value;
+  const onSearchClose = useCallback(() => {
+    setShowSearch(false);
+  }, []);
+
+  const onSearchOpen = useCallback(() => {
+    setShowSearch(true);
+  }, []);
+
+  const isSearchKeyboardEvent = useCallback(
+    (e: KeyboardEvent) => {
+      return (
+        e.type === 'keydown' &&
```

Comment:
We don't have this check in Web. Why is it needed?

Reply:
I just saw other spots in Connect focusing specifically on keydown, didn't think too much on it. I can remove as the method being called on `true` is idempotent and shouldn't matter on keyup/keydown or duplicates